src/target/riscv/riscv-013_reg.c

```diff
@@ -173,6 +174,64 @@ static int examine_vlenb(struct target *target)
 	return ERROR_OK;
 }
 
+enum misa_mxl {
+	MISA_MXL_INVALID = 0,
+	MISA_MXL_32 = 1,
+	MISA_MXL_64 = 2,
+	MISA_MXL_128 = 3
+};
+
+enum misa_mxl xlen_to_mxl(unsigned int xlen)
+{
+	switch (xlen) {
+	case 32:
+		return MISA_MXL_32;
+	case 64:
+		return MISA_MXL_64;
+	case 128:
+		return MISA_MXL_128;
+	}
+	return MISA_MXL_INVALID;
+}
+
+static int check_misa_mxl(const struct target *target)
+{
+	RISCV_INFO(r);
+
+	if (r->misa == 0) {
+		LOG_TARGET_WARNING(target, "'misa' register is read as zero."
+				"OpenOCD will not be able to determine some hart's capabilities.");
+		return ERROR_OK;
+	}
+	const unsigned int xlen = riscv_xlen(target);
+	assert(xlen <= sizeof(riscv_reg_t) * CHAR_BIT);
+	assert(xlen >= 2);
+	riscv_reg_t misa_mxl_field = (riscv_reg_t)0x3 << (xlen - 2);
```

Comment:
Details really:

- This can be const
- If you like, it can be called "mask" instead of "field" for more clarity

`const riscv_reg_t misa_mxl_mask = (riscv_reg_t)0x3 << (xlen - 2);`

Reply:
Addressed.